src/CoreFx.Private.TestUtilities/src/System/PlatformDetection.Unix.cs

```diff
@@ -103,6 +103,77 @@ private static DistroInfo ParseOsReleaseFile()
                     }
                 }
             }
+            else 
+            {
+                string fileName = null;
+                if (File.Exists("/etc/redhat-release"))
+                    fileName = "/etc/redhat-release";
+
+                if (fileName == null && File.Exists("/etc/system-release"))
+                    fileName = "/etc/system-release";
+                
+                if (fileName != null)
+                {
+                    // Parse the format like the following line:
+                    // Red Hat Enterprise Linux Server release 7.3 (Maipo)
+                    using (StreamReader file = new StreamReader(fileName))
+                    {
+                        string line = file.ReadLine();
+                        if (!String.IsNullOrEmpty(line))
+                        {
+                            if (line.StartsWith("Red Hat Enterprise Linux", StringComparison.OrdinalIgnoreCase))
+                            {
+                                ret.Id = "rhel";
+                            }
+                            else if (line.StartsWith("Centos", StringComparison.OrdinalIgnoreCase))
```

Comment:
Centos seems to successfully work using /etc/os-release, at least at https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster~2F/type/test~2Ffunctional~2Fcli~2F/build/20170905.03/workItem/System.Runtime.InteropServices.RuntimeInformation.Tests/wilogs I see 
`2017-09-05 16:38:03,416: INFO: proc(54): run_and_log_output: Output: Distro=centos VersionId=7 Pretty=CentOS Linux 7 (Core) Version=7 (Core) OS=Linux 3.10.0-514.26.2.el7.x86_64 #1 SMP Tue Jul 4 15:04:05 UTC 2017 OSVer=Unix 3.10.0.514 OSArch=X64 Arch=X64 Framework=.NET Core 4.6.25701.02`

Reply:
Yes, Centos currently work. but Centos is the future can decide to do the same as Red Hat. especially it is built from the same sources. currently, in Centos, it contains the same file we read the version info as the one in Red Hat. the current change will not affect the current behavior as we'll try to read the version on Centos the same way we do today. The change here is just a fallback path.

Reply:
sounds good